src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -50,6 +53,13 @@ private[avro] class AvroRelation(
     private val parameters: Map[String, String])
     (@transient val sqlContext: SQLContext) extends HadoopFsRelation with Logging {
 
+  // This UUID is used to avoid output file name collision between different appending write jobs.
+  // These jobs may belong to different SparkContext instances. Concrete data source implementations
+  // may use this UUID to generate unique file names (e.g., `part-r-<task-id>-<job-uuid>.parquet`).
+  //  The reason why this ID is used to identify a job rather than a single task output file is
+  // that, speculative tasks must generate the same output file name as the original task.
+  private val uniqueWriteJobId = UUID.randomUUID().toString
```

Comment:
The reason was that it seemed that the `BaseWriterContainer` was not setting the UUID so I was setting it myself.